magento/magento2
# Alternative fix for Multi Store Emails issue, Fix Async Emails issues, Fix Multiple Email issues.

### Description
This PR removes the previously introduced transportBuilderByStore class that was introduced in 2.2.4 to fix issue #11740 as this implementation caused unwanted regression/bug when sending multiple emails/async emails.
It adds new default parameter to setFrom method of transportBuilder class.

Possible sticking point, is this adds a new default parameter to public method of an API class.

### Fixed Issues (if relevant)
1. magento/magento2#11740:Sending emails from Admin in Multi-Store Environment defaults to Primary Store
2. magento/magento2#16165:Sales Emails Async Sending issue
3. magento/magento2#16203:New order notification is not sent to the admin
4. magento/magento2#14952:Confirmation emails have no FROM or FROM email address 2.2.4
5. magento/magento2#14945:Store Email Addresses are not used anymore
6. magento/magento2#16355:Magento 2.2.4 not sending from sales sender
7. magento/magento2#16509:Magento 2.2.4 Multishipping Order Confirmation Emails sent only for the first order
8. magento/magento2#14236:Sending multiple emails using the TransportBuilder class causes an Zend_Mail exception
9. magento/magento2#7739:"From Header set twice" when sending an email
10. magento/magento2#17781:Magento 2.2.5 Order Email Confirmation is not working

### Manual testing scenarios
Scenareo 1
- Single store
- Setup store email addresses
- Complete checkout
- Verify from address in confirmation email

Scenareo 2
- Multiple store
- Setup store email addresses for Store 1
- Setup store email addresses for Store 2
- Complete checkout on Store 1- Verify from address in confirmation email
- Complete checkout on Store 2- Verify from address in confirmation email

Scenareo 3
- Single Store
- Setup store email addresses
- Enable Async Email sending
- Disable cron job
- Complete checkout
- Complete checkout again
- Run cronjob once from command line
- Ensure both confirmation emails are received

Scenareo 4
- Single Store
- Setup store email addresses
- Add admin confirmation email address, using "separate email" as type.
- Complete checkout
- Ensure both confirmation and copy confirmation email are received.

### Contribution checklist
